src/Bus/TestBus.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Zenstruck\Messenger\Test\Bus;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class TestBus implements MessageBusInterface
+{
+    private MessageBusInterface $decorated;
+    /** @var list<Envelope> */
+    private array $messages = [];
```

Comment:
This is going to be a problem if the kernel is reset during the test. I think this data will need to be stored statically like we do with the TestTransport (and reset before each test).